nil pointer dereference on build-spec commmand

## Task summary
- Make sure that `./bin/gossamer build-spec` does not error like this

```
kishan@kishan-T480:~/code/gossamer$ ./bin/gossamer build-spec
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xedf993]

goroutine 1 [running]:
main.setDotGlobalConfigName(0xc0000da580, 0x0, 0xc00000c3c0)
	github.com/ChainSafe/gossamer/cmd/gossamer/config.go:531 +0x73
main.setDotGlobalConfig(0xee5120, 0x0, 0xc00000c3c0)
	github.com/ChainSafe/gossamer/cmd/gossamer/config.go:433 +0xb8
main.createBuildSpecConfig(0x118988e)
	github.com/ChainSafe/gossamer/cmd/gossamer/config.go:227 +0xec
main.buildSpecAction(0xc0000da580)
	github.com/ChainSafe/gossamer/cmd/gossamer/main.go:367 +0xfd
main.FixFlagOrder.func1(0xc0000da580)
	github.com/ChainSafe/gossamer/cmd/gossamer/flags.go:546 +0x48
github.com/urfave/cli.HandleAction({0xfb2fa0, 0xc000600940}, 0xa)
	github.com/urfave/cli@v1.22.5/app.go:524 +0xa8
github.com/urfave/cli.Command.Run({{0x1186a6a, 0xa}, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0x12945df, 0x40}, {0x0, ...}, ...}, ...)
	github.com/urfave/cli@v1.22.5/command.go:173 +0x652
github.com/urfave/cli.(*App).Run(0xc00097aa80, {0xc000032040, 0x2, 0x2})
	github.com/urfave/cli@v1.22.5/app.go:277 +0x705
main.main()
	github.com/ChainSafe/gossamer/cmd/gossamer/main.go:148 +0x35
```

## Specification reference
<!-- Provide a reference to the specification as to what is being implemented. -->
- 



## Other information and links
<!-- Add any other context, existing implementation reference or screenshots about the task here. -->
- 


<!-- Thank you 💪 -->

--- a/cmd/gossamer/config.go
+++ b/cmd/gossamer/config.go
@@ -43,19 +43,20 @@ var (
 // loadConfigFile loads a default config file if --chain is specified, a specific
 // config if --config is specified, or the default gossamer config otherwise.
 func loadConfigFile(ctx *cli.Context, cfg *ctoml.Config) (err error) {
-	if cfgPath := ctx.GlobalString(ConfigFlag.Name); cfgPath != "" {
-		logger.Info("loading toml configuration from " + cfgPath + "...")
-		if cfg == nil {
-			cfg = &ctoml.Config{} // if configuration not set, create empty configuration
-		} else {
-			logger.Warn(
-				"overwriting default configuration with id " + cfg.Global.ID +
-					" with toml configuration values from " + cfgPath)
-		}
-		return loadConfig(cfg, cfgPath) // load toml values into configuration
+	cfgPath := ctx.GlobalString(ConfigFlag.Name)
+	if cfgPath == "" {
+		return loadConfig(cfg, defaultGssmrConfigPath)
 	}
 
-	return loadConfig(cfg, defaultGssmrConfigPath)
+	logger.Info("loading toml configuration from " + cfgPath + "...")
+	if cfg == nil {
+		cfg = new(ctoml.Config)
+	} else {
+		logger.Warn(
+			"overwriting default configuration with id " + cfg.Global.ID +
+				" with toml configuration values from " + cfgPath)
+	}
+	return loadConfig(cfg, cfgPath)
 }
 
 func setupConfigFromChain(ctx *cli.Context) (*ctoml.Config, *dot.Config, error) {
@@ -212,7 +213,7 @@ func createImportStateConfig(ctx *cli.Context) (*dot.Config, error) {
 }
 
 func createBuildSpecConfig(ctx *cli.Context) (*dot.Config, error) {
-	var tomlCfg *ctoml.Config = new(ctoml.Config)
+	tomlCfg := new(ctoml.Config)
 	err := loadConfigFile(ctx, tomlCfg)
 	if err != nil {
 		logger.Errorf("failed to load toml configuration: %s", err)

--- a/cmd/gossamer/config.go
+++ b/cmd/gossamer/config.go
@@ -528,7 +528,7 @@ func setDotGlobalConfigName(ctx *cli.Context, tomlCfg *ctoml.Config, cfg *dot.Gl
 	}
 
 	// consider the name on config as a second priority
-	if tomlCfg.Global.Name != "" {
+	if tomlCfg != nil && tomlCfg.Global.Name != "" {
 		cfg.Name = tomlCfg.Global.Name
 		return nil
 	}

--- a/cmd/gossamer/config.go
+++ b/cmd/gossamer/config.go
@@ -43,7 +43,6 @@ var (
 // loadConfigFile loads a default config file if --chain is specified, a specific
 // config if --config is specified, or the default gossamer config otherwise.
 func loadConfigFile(ctx *cli.Context, cfg *ctoml.Config) (err error) {
-	// check --config flag and load toml configuration from config.toml
 	if cfgPath := ctx.GlobalString(ConfigFlag.Name); cfgPath != "" {
 		logger.Info("loading toml configuration from " + cfgPath + "...")
 		if cfg == nil {
@@ -53,12 +52,10 @@ func loadConfigFile(ctx *cli.Context, cfg *ctoml.Config) (err error) {
 				"overwriting default configuration with id " + cfg.Global.ID +
 					" with toml configuration values from " + cfgPath)
 		}
-		err = loadConfig(cfg, cfgPath) // load toml values into configuration
-	} else {
-		err = loadConfig(cfg, defaultGssmrConfigPath)
+		return loadConfig(cfg, cfgPath) // load toml values into configuration
 	}
 
-	return err
+	return loadConfig(cfg, defaultGssmrConfigPath)
 }
 
 func setupConfigFromChain(ctx *cli.Context) (*ctoml.Config, *dot.Config, error) {
@@ -215,15 +212,14 @@ func createImportStateConfig(ctx *cli.Context) (*dot.Config, error) {
 }
 
 func createBuildSpecConfig(ctx *cli.Context) (*dot.Config, error) {
-	var tomlCfg *ctoml.Config
-	cfg := &dot.Config{}
+	var tomlCfg *ctoml.Config = new(ctoml.Config)
 	err := loadConfigFile(ctx, tomlCfg)
 	if err != nil {
 		logger.Errorf("failed to load toml configuration: %s", err)
 		return nil, err
 	}
 
-	// set global configuration values
+	cfg := new(dot.Config)
 	if err := setDotGlobalConfig(ctx, tomlCfg, &cfg.Global); err != nil {
 		logger.Errorf("failed to set global node configuration: %s", err)
 		return nil, err
@@ -528,7 +524,7 @@ func setDotGlobalConfigName(ctx *cli.Context, tomlCfg *ctoml.Config, cfg *dot.Gl
 	}
 
 	// consider the name on config as a second priority
-	if tomlCfg != nil && tomlCfg.Global.Name != "" {
+	if tomlCfg.Global.Name != "" {
 		cfg.Name = tomlCfg.Global.Name
 		return nil
 	}

--- a/cmd/gossamer/main.go
+++ b/cmd/gossamer/main.go
@@ -80,9 +80,9 @@ var (
 		Description: "The build-spec command outputs current genesis JSON data.\n" +
 			"\tUsage: gossamer build-spec\n" +
 			"\tTo generate raw genesis file from default: " +
-			"gossamer build-spec --raw > genesis.json" +
+			"gossamer build-spec --raw --output genesis.json" +
 			"\tTo generate raw genesis file from specific genesis file: " +
-			"gossamer build-spec --raw --genesis genesis-spec.json > genesis.json",
+			"gossamer build-spec --raw --genesis genesis-spec.json --output genesis.json",
 	}
 
 	// importRuntime generates a genesis file given a .wasm runtime binary.

--- a/cmd/gossamer/main.go
+++ b/cmd/gossamer/main.go
@@ -401,9 +401,10 @@ func buildSpecAction(ctx *cli.Context) error {
 		if err != nil {
 			return fmt.Errorf("cannot write genesis spec file: %w", err)
 		}
+	} else {
+		fmt.Printf("%s\n", res)
 	}
 
-	fmt.Println(res)
 	return nil
 }
 

--- a/cmd/gossamer/main.go
+++ b/cmd/gossamer/main.go
@@ -397,11 +397,13 @@ func buildSpecAction(ctx *cli.Context) error {
 	}
 
 	if outputPath := ctx.String(OutputSpecFlag.Name); outputPath != "" {
-		return dot.WriteGenesisSpecFile(res, outputPath)
+		err = dot.WriteGenesisSpecFile(res, outputPath)
+		if err != nil {
+			return fmt.Errorf("cannot write genesis spec file: %w", err)
+		}
 	}
 
-	fmt.Printf("%s", res)
-
+	fmt.Println(res)
 	return nil
 }
 

--- a/cmd/gossamer/main.go
+++ b/cmd/gossamer/main.go
@@ -397,12 +397,10 @@ func buildSpecAction(ctx *cli.Context) error {
 	}
 
 	if outputPath := ctx.String(OutputSpecFlag.Name); outputPath != "" {
-		if err = dot.WriteGenesisSpecFile(res, outputPath); err != nil {
-			return err
-		}
-	} else {
-		fmt.Printf("%s", res)
+		return dot.WriteGenesisSpecFile(res, outputPath)
 	}
+
+	fmt.Printf("%s", res)
 
 	return nil
 }

--- a/cmd/gossamer/toml_config.go
+++ b/cmd/gossamer/toml_config.go
@@ -46,7 +46,7 @@ func loadConfig(cfg *ctoml.Config, fp string) error {
 		},
 	}
 
-	if err = tomlSettings.NewDecoder(file).Decode(&cfg); err != nil {
+	if err = tomlSettings.NewDecoder(file).Decode(cfg); err != nil {
 		logger.Errorf("failed to decode configuration: %s", err)
 		return err
 	}

